src/helpers.rs

```diff
@@ -64,7 +64,10 @@ fn type_hopefully_is(segments: &[syn::PathSegment], expected: &str) -> bool {
 }
 
 pub fn is_cow(segments: &[syn::PathSegment]) -> bool {
-    type_hopefully_is(segments, "std::borrow::Cow")
+    match segments.last() {
+        Some(syn::PathSegment { ident, .. }) => ident == "Cow",
+        _ => false,
+    }
```

Comment:
I wonder if this accidentially adds support for [`beef`](https://docs.rs/beef/latest/beef/).

Well, it's no problem. Before getting rid of the duck typing there is really no way to tell with #14.

Is it clear to you what is the `segments.last() == None` case? if not, could add `unreachable!()` or `segments.last().expect("type must always have at least one segment")`. Or maybe could use the error reporting facilities of `syn`, which haven't been used so far.

Reply:
Erroring here is not meaningful. If there is nothing, it will error downstream, so it's not wrong to use false.

Regarding beef, I suspect it will support beef. I wouldn't say this is bad; I think it's well-known to not name types after standard types unless it's planned for complete compatibility.

Reply:
@koivunej Gentle ping. Could you please take a look?